src/createUseState.lua

```diff
@@ -21,7 +21,9 @@ local function createUseState(component)
 		elseif value == NONE then
 			value = nil
 		end
-
+		
+		component.state[hookCount] = value
```

Comment:
Roact forbids setting state on render, which `createUseState` is called in. This is not a good solution.

Reply:
I know it doesn't *technically*, but it does fundamentally.